service-rpc/src/main/java/org/apache/iotdb/rpc/RpcUtils.java

```diff
@@ -244,7 +244,8 @@ public static String parseLongToDateWithPrecision(
           digits.insert(0, "0");
         }
       }
-      return datetime.substring(0, 19) + "." + digits + datetime.substring(19);
+      String timeZoneStr = datetime.substring(datetime.length() - 6);
+      return datetime.substring(0, datetime.length() - 6) + "." + digits + timeZoneStr;
```

Comment:
Hi, this PR looks good to me. 
I think we should also be careful when the timestampPrecision is us or ns.
Also, please check the failed tests. Thank you so much!

Reply:
ok，